src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -24,12 +29,114 @@ trait AttributeComponent {
     def name = column[String]("name")
     def valueString = column[Option[String]]("value_string")
     def valueNumber = column[Option[Double]]("value_number")
-    def valueDate = column[Option[Timestamp]]("value_date")
     def valueBoolean = column[Option[Boolean]]("value_boolean")
+    def valueEntityRef = column[Option[Long]]("value_entity_ref")
     def listIndex = column[Option[Int]]("list_index")
 
-    def * = (id, name, valueString, valueNumber, valueDate, valueBoolean, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+    def * = (id, name, valueString, valueNumber, valueBoolean, valueEntityRef, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+
+    def workspace = foreignKey("FK_ATTRIBUTE_ENTITY_REF", valueEntityRef, entityQuery)(_.id.?)
   }
 
-  protected val attributeQuery = TableQuery[AttributeTable]
+  protected object attributeQuery extends TableQuery(new AttributeTable(_)) {
+    def insertAttributeRecords(name: String, attribute: Attribute, workspaceId: UUID) = {
+      attribute match {
+        case AttributeEntityReferenceList(refs) => refs.zipWithIndex.map { case (ref, index) => insertAttributeRef(name, workspaceId, ref, Option(index))}
+        case AttributeValueList(values) =>
+          assertConsistentListMembers(values)
```

Comment:
Do we want a similar `assertConsistentEntityType` for `case AttributeEntityReferenceList`?  Is it OK to mix entity types in a reference list?